compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -1211,10 +1212,18 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 _ => false,
             };
             if can_derive {
+                let self_name = trait_pred.self_ty().to_string();
+                let self_span = self.tcx.def_span(adt.did);
+                if let Some(sym::Ord) = diagnostic_name {
```

Comment:
I think we should make an attempt at making this more generic than just `Ord` and `PartialOrd` - could you try [`rustc_middle::traits::util::supertraits`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/traits/util/fn.supertraits.html) to retrieve the supertraits? You might be able to call [`to_opt_poly_trait_ref`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/struct.Predicate.html#method.to_opt_poly_trait_ref) on `pred` to get the `PolyTraitRef` that you need.

Reply:
I implemented as you suggested and it caught a whole bunch of cases I missed!

I'm still a bit concerned about some traits having a Sized parent e.g. Clone
I'm not sure at what point the Sized traits get filtered out by the suggestions and if we can rely on that to occur in the future.
I guess thats why we have tests though!
Overall im happy with this approach.

Reply:
I think this is a good improvement and if we end up with some incorrect suggestions related to `Sized` then we can fix that in a follow-up.